yt_dlp/extractor/yandexvideo.py

```diff
@@ -268,35 +270,75 @@ class ZenYandexChannelIE(InfoExtractor):
         'url': 'https://zen.yandex.ru/tok_media',
         'info_dict': {
             'id': 'tok_media',
+            'description': 'md5:84d2fb177e478e44690f131805d7c842',
+            'title': 'ТОК',
         },
         'playlist_mincount': 169,
     }, {
         'url': 'https://zen.yandex.ru/id/606fd806cc13cb3c58c05cf5',
         'info_dict': {
             'id': '606fd806cc13cb3c58c05cf5',
+            'description': 'md5:517b7c97d8ca92e940f5af65448fd928',
+            'title': 'AcademeG DailyStream',
         },
         'playlist_mincount': 657,
     }]
 
-    def _entries(self, id, url):
-        webpage = self._download_webpage(url, id)
-        data_json = self._parse_json(re.findall(r'var\s?data\s?=\s?({.+?})\s?;', webpage)[-1], id)
-        for key in data_json.keys():
-            if key.startswith('__serverState__'):
-                data_json = data_json[key]
-        items = list(try_get(data_json, lambda x: x['feed']['items'], dict).values())
-        more = try_get(data_json, lambda x: x['links']['more']) or None
+    def _entries(self, channel_title, server_state_json, server_settings_json):
+        items = list(try_get(server_state_json, lambda x: x['feed']['items'], dict).values())
+
+        if(len(items) == 0):
+            items = list(try_get(server_settings_json, lambda x: x['exportData']['items']))
+
+        prev_next_page_id = None
+        more = try_get(
+            server_state_json, lambda x: x['links']['more']
+        ) or try_get(
+            server_settings_json, lambda x: x['exportData']['more']['link']) or None
+
+        if more:
+            next_page_id = try_get(parse_qs(more), lambda x: x['next_page_id'][0]) or None
+        else:
+            next_page_id = None
+
         for page in itertools.count(1):
-            for item in items:
+            for item in filter(lambda x: x.get('type') == 'gif', items):
                 video_id = item.get('publication_id') or item.get('publicationId')
                 video_url = item.get('link')
                 yield self.url_result(video_url, ie=ZenYandexIE.ie_key(), video_id=video_id.split(':')[-1])
-            if not more:
+            if (
+                not more
+                or len(items) == 0
+                or (next_page_id and next_page_id == prev_next_page_id)
+                or not next_page_id
+            ):
```

Comment:
not readable

Reply:
maybe GH fuckup, not sure

Reply:
Thanks for the review.
Can you offer a suggestion on how to make it readable? I'm a total newbie to Python so I'm not very familiar with the usual syntax.
In case GH is GitHub, and you're thinking that they reformatted it, no, that is how I wrote the code. In the languages I work with separating conditions on multiple lines with the operators aligned to the left is preferable but I will follow whatever convention the project follows, obviously, I'm just not familiar wiht it.

Reply:
yeah I can't think of a good way to improve this as it is.

One thing i would have done is, see what happens if i provide `None` as more to api.
in most cases api will either give null items or throw and error code. 
U can just use that to break instead of 4 diff cases.

ofc this will result in 1 extra useless api call but imo that would be more readable and generalized

Reply:
See if this new version is more readable. The problem with that code is that it has to be called during the next iteration not when more and the other variables are calculated.